pandas/core/generic.py

```diff
@@ -721,12 +721,37 @@ def transpose(self, *args, **kwargs):
         new_axes = self._construct_axes_dict_from(
             self, [self._get_axis(x) for x in axes_names]
         )
-        new_values = self.values.transpose(axes_numbers)
-        if kwargs.pop("copy", None) or (len(args) and args[-1]):
-            new_values = new_values.copy()
+
+        if (
+            self._is_homogeneous_type
+            and len(self._data.blocks)
+            and is_extension_array_dtype(self._data.blocks[0].dtype)
```

Comment:
we can avoid self._data references by making this `len(self.dtypes) and is_extension_array_dtype(self.dtypes.iloc[0])`

Reply:
ditto on 731 with self.dtypes